src/ContactList.js

```diff
@@ -1,21 +1,105 @@
 import React, { Component } from 'react';
-import App from './App';
-import ContactListPage from './ContactListPage';
 import ContactCard from './ContactCard';
+import { fromJS } from 'immutable';
+import { actions } from './redux/ContactList';
+import { connect } from 'react-redux';
 
 class ContactList extends Component {
-    constructor(props){
+    
+    constructor(props) {
         super(props);
-        this.state={contactItems: props.toItems }
+        this.state = { newItem: fromJS({}) , showAddContactCardFlag : false };
+        this.deleteCard = this.deleteCard.bind(this);
+        this.editCard =  this.editCard.bind(this);
+        this.addNewContactCard = this.addNewContactCard.bind(this);
+        this.addContactCard = this.addContactCard.bind(this);
+        this.showAddContactCardForm = this.showAddContactCardForm.bind(this);
+        this.resetForm = this.resetForm.bind(this);
     }
 
-     render(){
-         return(
-         this.state.contactItems.forEach(function(item) {
-                     <ContactCard contactData={ item } />
-         }, this)
-         );
- }
+    componentWillMount() {
+        this.props.initialList();
+    }
+
+    deleteCard(id) {
+        this.props.deleteContact({ id });
+    }   
+
+    editCard(id, newValue, field) {
+        this.props.editContact({id,newValue,field});
+    }
+
+    addNewContactCard(e, field) {
+        // this.props.newContact({ item: { [field]: e.target.value } });
+        const item = {};
+        item[field] = e.target.value;
+        console.log(item);
+        this.props.addNewContact( {item} )
```

Comment:
instead of 2 functions i should make it as one function.. and it should be called on button click???